### PR TITLE
[RFR] Refactored webpack configuration

### PR DESCRIPTION
**Pull request by djhi**
- [x] webpack config fully written in es6
- [x] Updated config to have frontend at http://localhost:8080 directly
- [x] Updated pm2 test config for e2e tests

### CODE REVIEW

File: webpack/loaders.js [hunk 1]
```diff
@@ -23,14 +13,18 @@ export default (appName) => {
...
+                'stage-0',
+            ],
         },
     }, {
         test: /\.json$/,
         loader: 'json-loader',
     }, {
         test: /\.jpe?g$|\.gif$|\.png$/,
-        loader: `url-loader?limit=10000&name=${appName}/[hash].[ext]`,
+        loader: `url-loader?limit=10000&name=/${appName}/[hash].[ext]`,
```
Comment (Kmaschta) [inline webpack/loaders.js]: Even for frontend app ?

File: webpack/loaders.js [hunk 2]
```diff
@@ -50,9 +44,9 @@ export default (appName) => {
         test: /\.html$/,
         loader: 'html',
     }, {
-        loader: ExtractTextPlugin.extract(`css!sass?${sassOptions}`),
+        loader: ExtractTextPlugin.extract('css!sass?' + sassOptions),
```
Comment (Kmaschta) [inline webpack/loaders.js]: Why no ES6?

File: webpack/loaders.js (hunk 1 above)
Comment (djhi) [inline webpack/loaders.js]: We have the same prb on zerodollar then. ping @jpetitcolas Help !

Comment (Kmaschta) [inline webpack/loaders.js]: We need to have `/[hash].[ext]` for the frontend and `/admin/[hash].[ext]` for admin.

Comment (djhi) [inline webpack/loaders.js]: Yes but that mean different loaders for different apps again

Comment (djhi) [inline webpack/loaders.js]: I wonder if there is a better option

Comment (jpetitcolas) [inline webpack/loaders.js]: Factorizing all loaders is perhaps not the good solution then.

Comment (djhi) [inline webpack/loaders.js]: what about this solution ?

File: webpack/admin.config.js [hunk 3]
```diff
@@ -17,10 +18,24 @@ module.exports = {
...
-    plugins: plugins('admin'),
+    plugins: [
+        definePlugin(),
+        extractTextPlugin('admin'),
+        new HtmlWebpackPlugin({
+            filename: 'admin/index.html',
+            template: `${__dirname}/../src/admin/index.html`,
+            chunks: ['index'],
+            hash: true,
+        }), new HtmlWebpackPlugin({
```
Comment (jpetitcolas) [inline webpack/admin.config.js]: I would put it on a new line to ease readability.

File: webpack/admin.config.js [hunk 4]
```diff
@@ -17,10 +18,24 @@ module.exports = {
...
     },
-    plugins: plugins('admin'),
+    plugins: [
+        definePlugin(),
+        extractTextPlugin('admin'),
+        new HtmlWebpackPlugin({
+            filename: 'admin/index.html',
+            template: `${__dirname}/../src/admin/index.html`,
+            chunks: ['index'],
+            hash: true,
```
Comment (jpetitcolas) [inline webpack/admin.config.js]: Not sure you need it with latest version of the plugin.

File: webpack/loaders.js [hunk 5]
```diff
@@ -2,15 +2,10 @@ import ExtractTextPlugin from 'extract-text-webpack-plugin';
...
-    const presets = [
-        'es2015',
-        'react',
-        'stage-0',
-    ];
-
-    if (process.env.NODE_ENV === 'development') {
-        presets.push('react-hmre');
+export default function (appName) {
+    let prefix = '';
```
Comment (jpetitcolas) [inline webpack/loaders.js]: Use a `const` and a ternary operator.

File: webpack/loaders.js [hunk 6]
```diff
@@ -23,14 +18,18 @@ export default (appName) => {
                 'transform-runtime',
                 'add-module-exports',
             ],
-            presets,
+            presets: [
```
Comment (jpetitcolas) [inline webpack/loaders.js]: Can't we use a `.babelrc` config file instead?

Comment (djhi) [inline webpack/loaders.js]: It has been problematic when we worked on a react-native project which required a specific babelrc to work. Having the conf in webpack allowed us to make all apps (react, ng-admin and react-native) work

File: webpack/admin.config.js (hunk 4 above)
Comment (djhi) [inline webpack/admin.config.js]: This is still mentioned on the documentation:
https://github.com/ampedandwired/html-webpack-plugin#configuration

Comment (jpetitcolas) [inline webpack/admin.config.js]: Yes, but it may be done automatically.

Comment (djhi) [inline webpack/admin.config.js]: Nope: https://github.com/ampedandwired/html-webpack-plugin/blob/master/index.js#L17

File: webpack/plugins.js [hunk 7]
```diff
@@ -1,37 +1,19 @@
-import config from 'config';
-import ExtractTextPlugin from 'extract-text-webpack-plugin';
-import HtmlWebpackPlugin from 'html-webpack-plugin';
-import webpack from 'webpack';
+import config from 'config'; // eslint-disable-line no-var
```
Comment (brikou) [inline webpack/plugins.js]: What is the purpose of `// eslint-disable-line no-var`?

Comment (djhi) [inline webpack/plugins.js]: None

Comment (djhi) [inline webpack/plugins.js]: left over from previous non es6 versions

Comment (djhi) [inline webpack/plugins.js]: Fixed

**Comment by fzaninotto** [comment]
The merge build fails. Can you try rebasing?

**Comment by djhi** [comment]
That fixed it :+1:

**Pull request merged**